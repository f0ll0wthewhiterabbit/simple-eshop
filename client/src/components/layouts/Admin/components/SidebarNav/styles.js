import styled from 'styled-components'
import { List, ListItemIcon, ListItemText, ListItem } from '@material-ui/core'
import blueGrey from '@material-ui/core/colors/blueGrey'
import { NavLink } from 'react-router-dom'

export const PagesList = styled(List)`
  padding-top: ${props => props.theme.spacing(2)}px;
  padding-bottom: ${props => props.theme.spacing(2)}px;
`

export const NavigationItem = styled(ListItem)`
  padding: 0;
`

export const NavigationLink = styled(NavLink)`
  padding: ${props => props.theme.spacing(1, 2)};
  width: 100%;
  display: flex;
  position: relative;
  text-align: left;
  align-items: center;
  justify-content: flex-start;
  text-decoration: none;
  color: ${blueGrey[500]};

  &.active {
    color: ${blueGrey[800]};
    text-decoration: underline;
  }
`

export const PageIcon = styled(ListItemIcon)`
  color: inherit;
  min-width: ${props => props.theme.spacing(5)}px;
`

export const PageText = styled(ListItemText)`
  color: inherit;

  .MuiListItemText-primary {
    font-weight: ${props => props.theme.typography.fontWeightMedium};
    font-size: 0.875rem;
  }
`
