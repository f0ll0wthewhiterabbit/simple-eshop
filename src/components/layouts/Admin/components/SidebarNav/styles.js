import styled from 'styled-components'
import { List, ListItemIcon, ListItemText } from '@material-ui/core'
import blueGrey from '@material-ui/core/colors/blueGrey'

const PagesList = styled(List)`
  padding-top: ${props => props.theme.spacing(2)}px;
  padding-bottom: ${props => props.theme.spacing(2)}px;
`

const PageIcon = styled(ListItemIcon)`
  color: ${blueGrey[800]};
  min-width: ${props => props.theme.spacing(5)}px;
`

const PageText = styled(ListItemText)`
  color: ${blueGrey[800]};

  .MuiListItemText-primary {
    font-weight: ${props => props.theme.typography.fontWeightMedium};
    font-size: 0.875rem;
  }
`

export { PagesList, PageIcon, PageText }
