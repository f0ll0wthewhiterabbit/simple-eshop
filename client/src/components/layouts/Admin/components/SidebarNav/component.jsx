import React from 'react'
import PropTypes from 'prop-types'
import PeopleIcon from '@material-ui/icons/People'
import ShoppingBasketIcon from '@material-ui/icons/ShoppingBasket'

import { ADMIN_USERS_PAGE_PATH, ADMIN_PRODUCTS_PAGE_PATH } from '../../../../../constants'
import { PagesList, NavigationLink, NavigationItem, PageIcon, PageText } from './styles'

const SidebarNav = ({ closeSidebar }) => {
  const pages = [
    {
      title: 'Users',
      href: ADMIN_USERS_PAGE_PATH,
      icon: <PeopleIcon />,
    },
    {
      title: 'Products',
      href: ADMIN_PRODUCTS_PAGE_PATH,
      icon: <ShoppingBasketIcon />,
    },
  ]

  const handleNavLinkClick = () => {
    closeSidebar()
  }

  return (
    <nav aria-label="navigation">
      <PagesList>
        {pages.map(page => (
          <NavigationItem key={page.title} button>
            <NavigationLink to={page.href} onClick={handleNavLinkClick} data-test="navigationLink">
              <PageIcon>{page.icon}</PageIcon>
              <PageText primary={page.title} />
            </NavigationLink>
          </NavigationItem>
        ))}
      </PagesList>
    </nav>
  )
}

SidebarNav.propTypes = {
  closeSidebar: PropTypes.func.isRequired,
}

export default SidebarNav
