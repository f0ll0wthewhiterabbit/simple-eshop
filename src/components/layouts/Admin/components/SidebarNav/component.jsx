import React from 'react'
import ListItem from '@material-ui/core/ListItem'
import PeopleIcon from '@material-ui/icons/People'
import ShoppingBasketIcon from '@material-ui/icons/ShoppingBasket'

import { PagesList, PageIcon, PageText } from './styles'

const SidebarNav = () => {
  const pages = [
    {
      title: 'Users',
      href: '/admin/users',
      icon: <PeopleIcon />,
    },
    {
      title: 'Products',
      href: '/admin/products',
      icon: <ShoppingBasketIcon />,
    },
  ]

  return (
    <nav aria-label="navigation">
      <PagesList>
        {pages.map(page => (
          <ListItem key={page.title} button>
            <PageIcon>{page.icon}</PageIcon>
            <PageText primary={page.title} />
          </ListItem>
        ))}
      </PagesList>
    </nav>
  )
}

export default SidebarNav
