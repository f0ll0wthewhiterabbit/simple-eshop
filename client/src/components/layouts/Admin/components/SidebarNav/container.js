import { connect } from 'react-redux'

import SidebarNav from './component'
import { closeSidebar } from '../../../../../store/actions'

const mapStateToProps = state => ({
  isSidebarOpened: state.getIn(['app', 'isSidebarOpened']),
})

const mapDispatchToProps = {
  closeSidebar,
}

export default connect(mapStateToProps, mapDispatchToProps)(SidebarNav)
