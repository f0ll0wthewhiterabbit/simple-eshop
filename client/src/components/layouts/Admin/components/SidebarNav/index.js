import SidebarNav from './container'

export default SidebarNav
