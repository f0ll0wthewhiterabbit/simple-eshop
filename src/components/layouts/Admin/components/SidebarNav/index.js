import SidebarNav from './component'

export default SidebarNav
