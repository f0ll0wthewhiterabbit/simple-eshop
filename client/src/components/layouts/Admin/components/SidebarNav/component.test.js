import React from 'react'
import { shallow } from 'enzyme'

import SidebarNav from './component'
import findByTestAttr from '../../../../../utils/findByTestAttr'

describe('SidebarNav component', () => {
  let wrapper
  const initialProps = {
    closeSidebar: jest.fn(),
  }
  const dataTestNavigationLink = 'navigationLink'

  const generateWrapper = passedProps => {
    const defaultProps = { ...initialProps }
    const props = { ...defaultProps, ...passedProps }

    return shallow(<SidebarNav {...props} />)
  }

  beforeEach(() => {
    wrapper = generateWrapper()
  })

  afterEach(() => {
    initialProps.closeSidebar.mockReset()
  })

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should close sidebar after clicking the link', () => {
    const firstNavigationLink = findByTestAttr(wrapper, dataTestNavigationLink).first()
    firstNavigationLink.simulate('click')

    expect(initialProps.closeSidebar).toHaveBeenCalledTimes(1)
  })
})
