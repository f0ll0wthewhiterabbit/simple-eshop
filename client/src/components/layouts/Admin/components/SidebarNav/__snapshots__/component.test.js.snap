// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarNav component should render correctly 1`] = `
<nav
  aria-label="navigation"
>
  <Styled(WithStyles(ForwardRef(List)))>
    <Styled(WithStyles(ForwardRef(ListItem)))
      button={true}
      key="Users"
    >
      <Styled(NavLink)
        data-test="navigationLink"
        onClick={[Function]}
        to="/admin/users"
      >
        <Styled(WithStyles(ForwardRef(ListItemIcon)))>
          <PeopleIcon />
        </Styled(WithStyles(ForwardRef(ListItemIcon)))>
        <Styled(WithStyles(ForwardRef(ListItemText)))
          primary="Users"
        />
      </Styled(NavLink)>
    </Styled(WithStyles(ForwardRef(ListItem)))>
    <Styled(WithStyles(ForwardRef(ListItem)))
      button={true}
      key="Products"
    >
      <Styled(NavLink)
        data-test="navigationLink"
        onClick={[Function]}
        to="/admin/products"
      >
        <Styled(WithStyles(ForwardRef(ListItemIcon)))>
          <ShoppingBasketIcon />
        </Styled(WithStyles(ForwardRef(ListItemIcon)))>
        <Styled(WithStyles(ForwardRef(ListItemText)))
          primary="Products"
        />
      </Styled(NavLink)>
    </Styled(WithStyles(ForwardRef(ListItem)))>
  </Styled(WithStyles(ForwardRef(List)))>
</nav>
`;
